Pull Request for 4/3 Code Review

mbhsueh opened the pull request.

mbhsueh committed:
base files for maingui

nerissalin committed:
Merge pull request #2 from nerissalin/maingui

base files for maingui

mbhsueh committed:
adjusted User Label and Button, removed User Info Panel; set preferred size for ConversationPanel

nerissalin committed:
Changed mainGUI layout - TODO: modularize out title panel

nerissalin committed:
modularized better

nerissalin committed:
cleaner title panel

mbhsueh committed:
Merge pull request #6 from nerissalin/SwiftFrame

Basic Frame and Layout for Swift GUI

mbhsueh committed:
Merge branch 'SwiftFrame' of https://github.com/nerissalin/codeu_project_2017 into userpanel

j-tang3 committed:
Merge pull request #4 from nerissalin/userpanel

adjusted User Label and Button, removed User Info Panel

vaage committed:
Made Uuid a class

Before Uuid was an interface, but every implementation was
the same. Every time someone made implemented Uuid, they
would have to call Uuids.complete on it to fill in the
missing functions.

Making Uuid a class solves all these problems as all the
methods will be implemented correctly and there only needs
to be one implementation.

Generators are still sepearate.

vaage committed:
Correcting import orders

vaage committed:
Reworking Test

Moved Test from common to Util.

Changed the implementation to use Date all the time and not just for
converting it to a string.

Added the first test for Time to ensure that "fromMs" and "toMs"
yielded the same values.

mbhsueh committed:
change colors to hex, added plus.png

vaage committed:
Merge pull request #18 from google/simplify_uuid

Changed Uuid from an interface to a class as all implementations were the same.

vaage committed:
Merge branch 'master' into move-time

vaage committed:
Update ServerFrontEnd.java

Caffeinix commented on src/codeu/chat/client/maingui/ChatMainGui.java:
```diff
@@ -0,0 +1,134 @@
+import codeu.chat.client.View;
+import codeu.chat.util.Logger;
+
+// Chat - top-level client application - Java Simple GUI (using Java Swing)
```
Please update this comment.

Caffeinix commented on src/codeu/chat/client/maingui/ChatMainGui.java:
```diff
@@ -0,0 +1,134 @@
+    mainFrame.setSize(790, 450);
+
+    // Main View - outermost graphics panel.
+    final JPanel mainViewPanel = new JPanel(new GridBagLayout());
```
I know this was copied from the simple GUI, but `GridBagLayout` is very complex, and it's usually quite difficult to get the layout properties you want by hand.  (It's really designed to be used by tools that create layouts graphically for you.)  I would consider either using a lot of simpler nested layout (like BorderLayouts and GridLayouts) or a third-party layout manager.  Most modern layout managers (including the ones for Android and iOS) are relative layout managers, which means you can specify constraints for each component based on the components around it.  If you search for `RelativeLayout Swing`, you'll find a bunch of these, which might work better.

Caffeinix commented on src/codeu/chat/client/maingui/TitlePanel.java:
```diff
@@ -0,0 +1,35 @@
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+
+// NOTE: JPanel is serializable, but there is no need to serialize MessagePanel
```
Update comment.

Caffeinix commented on src/codeu/chat/client/maingui/UserPanel.java:
```diff
@@ -0,0 +1,202 @@
+    titleLabelC.gridy = 0;
+    titleLabelC.anchor = GridBagConstraints.PAGE_START;
+
+    // final GridBagConstraints titleGapC = new GridBagConstraints();
```
Best not to check in dead code.  Just delete it.  It can always be retrieved later from version control if necessary.

Caffeinix commented on src/codeu/chat/client/maingui/UserPanel.java:
```diff
@@ -0,0 +1,202 @@
+
+    final JLabel titleLabel = new JLabel();
+    titleLabel.setBackground(new Color(239, 246, 247));
+    titleLabel.setFont(new Font("PT Serif", 0, 14)); // NOI18N
```
In general, you can't assume that this font will be available on all users' computers, since they may be using one of many different operating systems.

Caffeinix reviewed (approved):
Subtracting out the lines that have been copied verbatim from simplegui, the sum total of changes made here is less than 100 lines, and those have just been linear code that changes styles, no logic.  

I know that the learning curve has been steep (particularly with regard to version control), but successfully completing this project is going to require you to show a lot more than this in the coming weeks.  Remember that part of the point of this project is to demonstrate and develop your software engineering skills, and what I'm seeing here doesn't do either of those things.

Let's talk about what happened here during our meeting this evening.  I also have some suggestions for how we can improve this, which we can discuss then as well.

mbhsueh committed:
Merge pull request #3 from nerissalin/release

Merge upstream/release into release

mbhsueh committed:
Merge branch 'master' into userpanel

mbhsueh committed:
Merge pull request #7 from nerissalin/userpanel

change colors to hex, added plus.png

vaage committed:
Merge pull request #19 from google/move-time

Reworking Test

vaage committed:
Exposing params in run scripts

Before the run script had defaults for most values. This
starting causing problems when hosting the server as it
would require the run scripts to change.

Now all the paramters are passed in when calling the run
script. If any value is missing, the script exits and a
usage message is printed.

vaage committed:
Merge pull request #30 from google/update-run-scripts

Exposing params in run scripts

vaage committed:
Added Optional Param for Relay Address

The java code supports an optional relay address but
it was never passed through. This change adds support
to the run_server script to pass through an optional
relay address.

vaage committed:
Merge pull request #32 from google/update-run-server

Added Optional Param for Relay Address

mbhsueh committed:
removed dead code and updated comments

mbhsueh committed:
users can also sign in by double clicking on their username

mbhsueh committed:
Merge remote-tracking branch 'upstream/master'

j-tang3 committed:
Merge pull request #9 from nerissalin/userpanel

removed dead code and updated comments

j-tang3 committed:
Merge pull request #10 from nerissalin/mouseinput

users can also sign in by double clicking on their username

nerissalin committed:
Allows users to type into a text box and also use enter to send

nerissalin committed:
Removed Unnecessary Imports

j-tang3 committed:
Merge pull request #11 from nerissalin/EnterMeMessages

Enter Messages
